lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -346,81 +372,65 @@ def main():
-                                           all_databases,
-                                           login_port, config_file,
-                                           socket, ssl_cert, ssl_key, ssl_ca)
-            if rc != 0:
-                module.fail_json(msg="%s" % stderr)
-            else:
-                module.exit_json(changed=True, db=db_name, msg=stdout)
 
-        elif state == "present":
-            module.exit_json(changed=False, db=db_name)
+    existence_list = []
+    non_existence_list = []
 
-    else:
-        if state == "present":
-            if module.check_mode:
-                changed = True
-            else:
-                try:
-                    changed = db_create(cursor, db, encoding, collation)
-                except Exception as e:
-                    module.fail_json(msg="error creating database: %s" % to_native(e),
-                                     exception=traceback.format_exc())
-            module.exit_json(changed=changed, db=db_name)
-
-        elif state == "import":
-            if module.check_mode:
-                module.exit_json(changed=True, db=db_name)
+    if not all_databases:
+        for each_database in db:
+            if db_exists(cursor, [each_database]):
+                existence_list.append(each_database)
             else:
-                try:
-                    changed = db_create(cursor, db, encoding, collation)
-                    if changed:
-                        rc, stdout, stderr = db_import(module, login_host, login_user,
-                                                       login_password, db, target, all_databases,
-                                                       login_port, config_file, socket, ssl_cert, ssl_key, ssl_ca)
-                        if rc != 0:
-                            module.fail_json(msg="%s" % stderr)
-                        else:
-                            module.exit_json(changed=True, db=db_name, msg=stdout)
-                except Exception as e:
-                    module.fail_json(msg="error creating database: %s" % to_native(e),
-                                     exception=traceback.format_exc())
-
-        elif state == "absent":
-            module.exit_json(changed=False, db=db_name)
-
-        elif state == "dump":
-            if module.check_mode:
-                module.exit_json(changed=False, db=db_name)
-            module.fail_json(msg="Cannot dump database %r - not found" % (db_name))
+                non_existence_list.append(each_database)
+
+    if state == "absent":
+        if module.check_mode:
+            module.exit_json(changed=bool(existence_list), db=db_name, db_list=db)
+        try:
+            changed = db_delete(cursor, existence_list)
+        except Exception as e:
+            module.fail_json(msg="error deleting database: %s" % to_native(e))
+        module.exit_json(changed=changed, db=db_name, db_list=db)
```

Comment:
it's repeated at the and of each condition. Maybe we could use it one time after
```suggestion
        module.exit_json(changed=changed, db=db_name, db_list=db)
```

Reply:
Actually `exit_json` is not same for all 4 state's. Its same for state=absent/present or state=dump/import.